src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -198,21 +203,47 @@ protected void install() throws InterruptedException {
                 }
             }
 
+            Topics deviceGroupTopics = this.config.lookupTopics(CONFIGURATION_CONFIG_KEY, DEVICE_GROUPS_TOPICS);
+            Topics certAuthorityTopic = this.config.lookupTopics(CONFIGURATION_CONFIG_KEY, CERTIFICATE_AUTHORITY_TOPIC);
+            String caKeyPath = Coerce.toString(certAuthorityTopic.findOrDefault("", CA_PRIVATE_KEY_URI));
+            String caCertPath = Coerce.toString(certAuthorityTopic.findOrDefault("", CA_CERTIFICATE_URI));
+            Topic caType = certAuthorityTopic.lookup(CA_TYPE_TOPIC);
+
+
             if (whatHappened == WhatHappened.initialized || node == null) {
                 updateDeviceGroups(whatHappened, deviceGroupTopics);
-                updateCAType(caTypeTopic);
+                updateCertificateAuthority();
             } else if (node.childOf(DEVICE_GROUPS_TOPICS)) {
                 updateDeviceGroups(whatHappened, deviceGroupTopics);
-            } else if (node.childOf(CA_TYPE_TOPIC)) {
-                if (caTypeTopic.getOnce() == null) {
+            } else if (node.childOf(CERTIFICATE_AUTHORITY_TOPIC)) {
+                // If the CA type is null because of the change to its configuration to null, do not update the CA.
```

Comment:
I'd argue this is wrong. A customer should be able to know the type of CA by looking at the configuration, without needing to be aware of previous state. Make this deterministic.